x-pack/elastic-agent/pkg/agent/cmd/container.go

```diff
@@ -112,15 +116,50 @@ all the above actions will be skipped, because the Elastic Agent has already bee
 occurs on every start of the container set FLEET_FORCE to 1.
 `,
 		Run: func(c *cobra.Command, args []string) {
-			if err := containerCmd(streams, c, flags, args); err != nil {
+			iscloud, err := c.Flags().GetBool("cloud")
+			if err != nil {
+				fmt.Fprintf(streams.Err, "Error: %v\n", err)
+				os.Exit(1)
+			}
+			if iscloud {
+				err = containerCloudCmd(streams, c, flags, args)
+			} else {
+				err = containerCmd(streams, c, flags, defaultAccessConfig())
+			}
+			if err != nil {
 				fmt.Fprintf(streams.Err, "Error: %v\n", err)
 				os.Exit(1)
 			}
 		},
 	}
+	cmd.Flags().Bool("cloud", false, "Start container in cloud mode")
+	return &cmd
+}
+
+func containerCloudCmd(streams *cli.IOStreams, cmd *cobra.Command, flags *globalFlags, args []string) error {
+	log, err := logger.New("bootstrap")
+	if err != nil {
+		return err
+	}
+	log.Info("Starting Elastic Agent container in cloud mode")
+	// run legacy APM Server if APM_SERVER_PATH is given
+	if apmPath := os.Getenv("APM_SERVER_PATH"); apmPath != "" {
+		if err := runLegacyAPMServer(log, apmPath, args); err != nil {
+			return err
+		}
+	}
+	// setup and start Fleet Server respecting configuration files
+	cfg := defaultAccessConfig()
+	if err := readYaml(filepath.Join(paths.Config(), "fleet-setup.yml"), &cfg); err != nil {
+		return errors.New(err, "parsing fleet-setup.yml")
+	}
+	if err := readYaml(filepath.Join(paths.Config(), "credentials.yml"), &cfg); err != nil {
+		return errors.New(err, "parsing credentials.yml")
+	}
+	return containerCmd(streams, cmd, flags, cfg)
```

Comment:
When any of these return, won't the apm-server process become orphaned and adopted by init? I can see where you kill elastic-agent when apm-server exits, but not the other way around.

Reply:
When the `containerCmd` returns, the whole container exits, therefore the APM Server would also be shutdown. But I see your point, and will explicitly terminate.